core/src/main/kotlin/io/rover/campaigns/core/data/graphql/operations/data/DeviceContext.kt

```diff
@@ -98,7 +100,8 @@ internal fun DeviceContext.Companion.decodeJson(json: JSONObject, dateFormatting
         location = json.optJSONObject("location").whenNotNull { locationJson ->
             Location.decodeJson(locationJson, dateFormatting)
         },
-        advertisingIdentifier = json.safeOptString("advertisingIdentifier")
+        advertisingIdentifier = json.safeOptString("advertisingIdentifier"),
+        conversions = json.optJSONArray("conversions").getStringIterable().toList()
```

Comment:
This will cause an NPE if `json.optJSONArray("conversions")` returns null. The `optJSONArray` can return null but as it's Java and isn't annotated with `@Nullable` or `@NotNull` it returns what is called a platform type. There's no compile time restrictions around platform types as the kotlin compiler can't infer whether it's nullable or not, we can treat them as nullable or non-null.

You probably wanna rewrite as this if there's a chance `json.optJSONArray("conversions")` can return null
`json.optJSONArray("conversions")?.getStringIterable()?.toList() ?: emptyList()`

Reply:
curious why is there a need to decode the device context? We never send it back to the device

Reply:
nvm I see its used when restoring events from disk